src/EffectiveCSharp.Analyzers/FormattableStringForCultureSpecificStringsAnalyzer.cs

```diff
@@ -0,0 +1,88 @@
+﻿namespace EffectiveCSharp.Analyzers;
+
+/// <summary>
+/// A <see cref="DiagnosticAnalyzer"/> for Effective C# Item #5 - Use FormattableString for culture specific strings
+/// </summary>
+/// <seealso cref="Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer" />
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class FormattableStringForCultureSpecificStringsAnalyzer : DiagnosticAnalyzer
+{
+    private static readonly LocalizableString Title = "Prefer FormattableString for culture-specific strings";
+    private static readonly LocalizableString MessageFormat = "Use 'FormattableString' instead of 'string' for culture-specific interpolated strings";
```

Comment:
Actually parameterize this because the guidance varies based on language version